yt_dlp/extractor/dplay.py

```diff
@@ -369,6 +369,174 @@ def _real_extract(self, url):
         return self._get_disco_api_info(url, self._match_id(url), **self._DISCO_API_PARAMS)
 
 
+class GoDiscoveryIE(DiscoveryPlusBaseIE):
+    _VALID_URL = r'https?://(?:go\.)?discovery\.com/video' + DPlayBaseIE._PATH_REGEX
```

Comment:
Add a test without `go.`. `only_matching` test is sufficient. Same for others

Reply:
@Sipherdrakon Please do this. The PR is stuck as pending-fixes because of potential URL conflict, which adding this test successfully would resolve